datastream/dataset.py

```diff
@@ -340,6 +329,32 @@ def group_split(
             ).items()
         }
 
+
+    def new_columns(
+        self: Dataset[T], **kwargs: Callable[pd.Dataframe, pd.Series]
+    ) -> Dataset[T]:
+        '''
+        Append new column(s) to the :attr:`.Dataset.dataframe` by passing the
+        new column names as keywords with functions that take the
+        :attr:`.Dataset.dataframe` as input and return :func:`pandas.Series`.
+
+        >>> (
+        ...     Dataset.from_dataframe(pd.DataFrame(dict(number=[1, 2, 3])))
+        ...     .new_columns(twice=lambda df: df['number'] * 2)
+        ...     .map(lambda row: row['twice'])
+        ... )[-1]
+        6
+        '''
+        if len(set(kwargs.keys()) & set(self.dataframe.columns)) >= 1:
+            raise ValueError('Should not replace existing columns')
+
+        dataframe = self.dataframe.assign(**kwargs)
+        return Dataset(
+            dataframe=dataframe,
+            length=len(dataframe),
```

Comment:
We assume that this function does not change the length of the dataset right? So 
```
Dataset(
    dataframe=self.dataframe.assign(...),
    length=len(self),
    ...
)
```
should be just fine?

Reply:
I prefer to do the same as we do in `from_dataframe`